### PR TITLE
full screen effects and disintegration, bubbles, and snow effects

### PR DESCRIPTION
**Pull request by jwunderl**
## normal effects

minor cleanup throughout, replace old tests

'ashes' / disintegration effect

## Full screen effects

Subclass of ParticleEffect that can occur over entire screen as background / foreground effects

### CODE REVIEW

**Comment by pelikhan** [comment]
Background effect? Looks amazing!

**Comment by jwunderl** [comment]
I think background effect does works better. Only a minor chance of misinterpreting it to have a correlation to the z index (e.g. background as opposed to foreground), but I think that's reasonable / easy to see when it occurs and fix as needed

**Comment by pelikhan** [comment]
Scene effects?

File: libs/particles/effects.ts [hunk 1]
```diff
@@ -53,32 +53,32 @@ namespace particles {
...
+    export class FullScreenEffect extends ParticleEffect {
         protected source: ParticleSource;
 
         constructor(sourceFactory: (anchor: ParticleAnchor, particlesPerSecond: number) => ParticleSource) {
             super(sourceFactory);
         }
 
         /**
-         * Creates a new Background Effect that occurs over entire screen
+         * Creates a new effect that occurs over theentire screen
```
Comment (pelikhan) [inline libs/particles/effects.ts]: typo

File: libs/particles/factories.ts [hunk 2]
```diff
@@ -368,37 +368,50 @@ namespace particles {
         }
     }
 
+    interface ColorCount {
```
Comment (pelikhan) [inline libs/particles/factories.ts]: use a class

**Comment by jwunderl** [comment]
As a side note, it's fairly trivial to add a true 'disintegration effect' (where the pixels that get chosen for the effect are removed from the source if possible) to the ashes effect, and it does look pretty cool. it would be a really quick way to run out of memory on hardware though. Still fun:

**Comment by pelikhan** [comment]
Even if you make it faster and reduce the particle lifespan?

**Comment by pelikhan** [comment]
is this tested on HW?

**Comment by jwunderl** [comment]
Issue I was thinking of was that it would have to clone the image each time the effect is used, to 0 out the selected pixels. I ran all the games that were gifs in the original post, I'll test the variant where it removes the selected pixels and see if it does actually work

**Comment by jwunderl** [comment]
Hm, just noticed hardware doesn't initialize number array contents to 0 instead of undefined like the browser does, have to be careful about that. Also, there is no array.fill which is useful when initializing to 0 isn't handled by default.

**Comment by pelikhan** [comment]
File bugs on missing array functions. Slowly adding them.

Get Outlook for iOS<https://aka.ms/o0ukef>

________________________________

**Comment by pelikhan** [comment]
Ready to go?

**Comment by jwunderl** [comment]
Yup!

File: libs/particles/effects.ts [hunk 3]
```diff
@@ -23,20 +23,86 @@ namespace particles {
         }
     }
 
+    /**
+     * Anchor used for effects that occur across the screen.
+     */
+    class FullScreenAnchor implements ParticleAnchor {
```
Comment (pelikhan) [inline libs/particles/effects.ts]: SceneAnchor

**Pull request merged**